packages/experiments/src/components/FloatingSuggestions/Suggestions/Suggestions.types.ts

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { ISuggestionModel } from 'office-ui-fabric-react/lib/Pickers';
+import { ISuggestionItemProps } from './SuggestionsItem.types';
+import { IPersonaProps } from 'office-ui-fabric-react/lib/Persona';
+import { IRefObject } from 'office-ui-fabric-react/lib/Utilities';
+
+export interface ISuggestionsCoreProps<T> extends React.ClassAttributes<any> {
+  /**
+   * Gets the component ref.
+   */
+  componentRef?: IRefObject<{}>;
+  /**
+   * How the suggestion should look in the suggestion list.
+   */
+  onRenderSuggestion: ISuggestionItemProps<T>['onRenderSuggestion'];
+  /**
+   * What should occur when a suggestion is clicked
+   */
+  onSuggestionClick: (ev: React.MouseEvent<HTMLElement>, item: T, index: number) => void;
+  /**
+   * The list of Suggestions that will be displayed
+   */
+  suggestions: ISuggestionModel<T>[];
+  /**
+   * Function to fire when one of the optional remove buttons on a suggestion is clicked.
+   */
+  onSuggestionRemove?: (ev: React.MouseEvent<HTMLElement>, item: IPersonaProps, index: number) => void;
+  /**
+   * Screen reader message to read when there are suggestions available.
+   */
+  suggestionsAvailableAlertText?: string;
+  /**
+   * An ARIA label for the container that is the parent of the suggestions.
+   */
+  suggestionsContainerAriaLabel?: string;
+  /**
+   * the classname of the suggestionitem.
+   */
+  suggestionsItemClassName?: string;
+  /**
+   * Maximum number of suggestions to show in the full suggestion list.
+   */
+  resultsMaximumNumber?: number;
+  /**
+   * Indicates whether to show a button with each suggestion to remove that suggestion.
+   */
+  showRemoveButtons?: boolean;
+  /**
+   * Indicates whether to loop around to the top or bottom of the the suggestions
+   * on calling nextSuggestion and previousSuggestion, respectively
+   */
+  shouldLoopSelection: boolean;
+}
+
+export interface ISuggestionsControlProps<T> extends React.ClassAttributes<any>, ISuggestionsCoreProps<T> {
+  /**
+   * An ARIA label for the container that is the parent of the suggestions header items.
+   */
+  suggestionsHeaderContainerAriaLabel?: string;
+  /**
+   * An ARIA label for the container that is the parent of the suggestions footer items.
+   */
+  suggestionsFooterContainerAriaLabel?: string;
+  /**
+   * The header items props
+   */
+  headerItemsProps?: ISuggestionsHeaderFooterProps[];
+  /**
+   * The footer items props
+   */
+  footerItemsProps?: ISuggestionsHeaderFooterProps[];
+  /**
+   * Whether or not the first selectable item in the suggestions list should be selected
+   */
+  shouldSelectFirstItem?: () => boolean;
+  /**
+   * The CSS classname of the suggestions list.
+   */
+  className?: string;
+  /**
+   * Completes the suggestion
+   */
+  completeSuggestion: () => void;
```

Comment:
This is part of the copypasta'd suggestions code. It looks like from usage the name is better as `onCurrentlySelectedSuggestionChosen`, since it's called when the user picks an item that isn't a header or footer item.. but rather than pass that item back, it relies on the callee to know what the selected item is.

I'll rename it in this PR but I hope we can get rid of this pattern when we rewrite suggestions.